code/game/objects/structures/tank_dispenser.dm

```diff
@@ -24,19 +24,26 @@
 
 /obj/structure/dispenser/New()
 	. = ..()
+	while(oxygentanks>0)
```

Comment:
Can't you make tank dispensers with like plasteel or something? What happens to those?

Also, use a for loop.